guides/internals.rst

```diff
@@ -0,0 +1,342 @@
+===================================
+How to Work on Hypothesis Internals
+===================================
+
+This is a guide to how to work on Hypothesis internals,
+with a particular focus on helping people who are new to it.
+Right now it is very rudimentary and is intended primarily for people who are
+looking to get started writing shrink passes as part of our `current outreach
+program to get more people doing that <https://github.com/HypothesisWorks/hypothesis-python/issues/1093>`_,
+but it will expand over time.
+
+------------------------
+Bird's Eye View Concepts
+------------------------
+
+The core engine of Hypothesis is called Conjecture.
+
+The "fundamental idea" of Conjecture is that you can represent an arbitrary
```

Comment:
I would reverse these, so we use a PRNG and view that as bytes:

> The "fundamental idea" of Conjecture is that you can represent an arbitrary randomized test case as a sequence of outputs from some pseudo-random number generator (PRNG), which in turn can be viewed as a string of bytes representing the underlying entropy.  Whenever you want to do something "random", you read the next bytes and do what they tell you to do.

Regardless, delete the word "basically".